firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/KeyValueBuilder.kt

```diff
@@ -17,23 +17,43 @@
 package com.google.firebase.crashlytics
 
 /** Helper class to enable fluent syntax in [setCustomKeys] */
-class KeyValueBuilder(private val crashlytics: FirebaseCrashlytics) {
+@Suppress("DEPRECATION")
+class KeyValueBuilder(
+  // TODO(mrober): Remove this param and make ctor internal in 2025.
+  @Deprecated(message = "The crashlytics instance is no longer needed and will be removed in 2025.")
+  private val crashlytics: FirebaseCrashlytics? = null
+) {
```

Comment:
If we're doing a deprecation, I think we should probably deprecate the `KeyValueBuilder` class altogether, in favor of `CustomKeysAndValues.Builder`. For these reasons:

1. They both provide the same functionality, and expose the same functions. 
2. The `KeyValueBuilder`'s `key()` function takes a parameter named `key`, which sounds repetitive when calling it with named arguments:
    ```kotlin
    crashlytics.setCustomKeys {
      key(key = "foo", value = "bar")
    }
    ```

If we do deprecate this class, we'd also have to change the KTX API surface from the existing `fun FirebaseCrashlytics.setCustomKeys(init: KeyValueBuilder.() -> Unit)` to `fun FirebaseCrashlytics.setCustomKeys(init: CustomKeysAndValues.Builder.() -> Unit)` and then provide some workarounds to retain source compatibility. (I drafted the changes [here](https://github.com/firebase/firebase-android-sdk/compare/main...rpf-crashlytics-ktx-customkeys-draft))

It's better for readability (although it doesn't support named arguments since `CustomKeysAndValues` is written in Java)
```kotlin
crashlytics.setCustomKeys {
  putString("foo", "bar")
}
```

Reply:
I'm not sure we should be deprecating `KeyValueBuilder` immediately.

Potential alternatives:
- `CustomKeysAndValues.Builder`
- https://github.com/firebase/firebase-android-sdk/commit/58d6e6e0f67c70c203190a103f338d5f94fcd526

Additionally, while changing the parameter type of `init` might seem like a no-op, it *very likely* is considered a breaking change requiring a major version bump on SDK.

Reply:
Agreed it'll be a breaking change. We should indeed do it at a major version bump if we decide to take that route.

Reply:
I like these but I will write it up in a 1 pager and we can look at it later for a breaking change. For now, this PR will not break any public APIs.

Also what do you think of `set` instead of `putX`? Then we can say like `{ this["key"] = "value" }`